src/compiler/interface.jl

```diff
@@ -171,6 +173,39 @@ function copy!(x::AbstractVector,  gs::Grads)
   x
 end
 
+# # Gradient Algebra: unary 
+# for op in (:+, :-)
+#   @eval broadcasted(::typeof($op), gs::Grads) = map($op, gs)
+# end
+
+# # Gradient Algebra: binary 
+# for op in (:+, :-)
+#   @eval broadcasted(::typeof($op), gs1::Grads, gs2::Grads) = map($op, gs1, gs2)
+# end
+
+broadcasted(f, gss::Grads...) = map(f, gss...)
+broadcasted(f, gs1::Grads, xs...) = map(f, gs1, xs...)
+broadcasted(::typeof(*), a::Number, gs::Grads) = map(*, gs, a)
```

Comment:
These seem useful but I'm worried about the ambiguity and the asymmetry. Since `Grads` behaves like a `Dict` and not an `AbstractArray`, it's unclear what the correct broadcasting semantics are.

Reply:
I'd like to support at least `0.1 .* gs1  .+ 0.9 .* gs2`, can remove more general broadcast